flixel/graphics/frames/bmfont/BMFontChar.hx

```diff
@@ -0,0 +1,143 @@
+package flixel.graphics.frames.bmfont;
+
+import haxe.io.BytesInput;
+import haxe.xml.Access;
+import UnicodeString;
+
+using StringTools;
+
+/**
+ * Bitmap font character data used internally via `FlxBitmapFont.fromAngelCode` to serialize text,
+ * xml or binary files exported from [BMFont](https://www.angelcode.com/products/bmfont/)
+ * 
+ * @since 5.6.0
+ * @see [flixel.graphics.frames.FlxBitmapFont.fromAngelCode](https://api.haxeflixel.com/flixel/graphics/frames/FlxBitmapFont.html#fromAngelCode)
+ */
+@:structInit
+@:allow(flixel.graphics.frames.bmfont.BMFont)
+class BMFontChar
+{
+	public var id:Int;
+	public var x:Int;
+	public var y:Int;
+	public var width:Int;
+	public var height:Int;
+	public var xoffset:Int;
+	public var yoffset:Int;
+	public var xadvance:Int;
+	public var page:Int;
+	public var chnl:Int;
+	public var letter:Null<String> = null;
+	
+	static inline function fromXml(charNode:Access):BMFontChar
+	{
+		return {
+			id: Std.parseInt(charNode.att.id),
+			x: Std.parseInt(charNode.att.x),
+			y: Std.parseInt(charNode.att.y),
+			width: Std.parseInt(charNode.att.width),
+			height: Std.parseInt(charNode.att.height),
+			xoffset: (charNode.has.xoffset) ? Std.parseInt(charNode.att.xoffset) : 0,
+			yoffset: (charNode.has.yoffset) ? Std.parseInt(charNode.att.yoffset) : 0,
+			xadvance: (charNode.has.xadvance) ? Std.parseInt(charNode.att.xadvance) : 0,
+			page: Std.parseInt(charNode.att.page),
+			chnl: Std.parseInt(charNode.att.chnl),
+			letter: charNode.has.letter ? charNode.att.letter : null
+		};
+	}
+	
+	static function listFromXml(charsNode:Access):Array<BMFontChar>
+	{
+		final chars = charsNode.nodes.char;
+		return [ for (char in chars) fromXml(char) ];
+	}
+	
+	static function fromText(kerningText:String):BMFontChar
+	{
+		var id:Int = -1;
+		var x:Int = -1;
+		var y:Int = -1;
+		var width:Int = 0;
+		var height:Int = 0;
+		var xoffset:Int = 0;
+		var yoffset:Int = 0;
+		var xadvance:Int = 0;
+		var page:Int = -1;
+		var chnl:Int = -1;
+		var letter:String = null;
```

Comment:
> this isn't part of the spec
what does "this" refer to?

Reply:
https://www.angelcode.com/products/bmfont/doc/file_format.html

Didn't realize this wasn't already in the PR description or something like that

Reply:
And I was specifically calling out that `letter` isn't in the spec, in case that's what you were asking.

Reply:
Yeah I found that the `letter` field gets generated sometimes for the text/xml outputs (despite not being part of the spec)  so I thought of adding it here just in case

Reply:
I'm not sure why and when `letter` appears, but it does, so I figured it should be in the type as an optional field